src/MicrosoftTeams.ts

```diff
@@ -21,14 +21,26 @@ namespace microsoftTeams {
 
   const version = "1.2";
 
-  const validOrigins = [
-    "https://teams.microsoft.com",
-    "https://teams.microsoft.us",
-    "https://int.teams.microsoft.com",
-    "https://devspaces.skype.com",
-    "https://ssauth.skype.com",
-    "http://dev.local" // local development
-  ];
+  const validHostRegEx = new RegExp(
+    "https://teams.microsoft.com" +
+    "|" +
+    "https://teams.microsoft.us" +
+    "|" +
+    "https://int.teams.microsoft.com" +
+    "|" +
+    "https://devspaces.skype.com" +
+    "|" +
+    "https://ssauth.skype.com" +
+    "|" +
+    "http://dev.local" + // local development
+    "|" +
+    "https://msft.spoppe.com" +
+    "|" +
+    "http*.*sharepoint.com" +
```

Comment:
This looks wrong. I think this clause of the regexp means
    `"htt" + (zero or more "p"s) + (zero or more of any characters) + "sharepoint" + (any character) + com`

which would match the following:
- htt://x.phishing-site-sharepoint.com
- httpppppp://x.malicious-sharepoint.com
- httpppppp://x.malicious-sharepoint-com
- httpmalicious-sharepointxcom

Reply:
Consider re-using the logic we have for domain validation in extensionTabService.ts.

Reply:
updated with domain validation logic in extensiontabService